runtime/compare.c

```diff
@@ -107,20 +109,56 @@ static intnat compare_val(value v1, value v2, int total)
 #define GREATER 1
 #define UNORDERED ((intnat)1 << (8 * sizeof(value) - 1))
 
+static bool compare_poll_actions(struct compare_item** sp,
+                                 struct compare_stack* stk,
+                                 value *orig1, value *orig2,
+                                 value *v1, value *v2)
+{
+  value o1 = *orig1, o2 = *orig2;
+  if (caml_check_pending_actions()) {
+    compare_free_stack(stk);
+    CAMLparam2(o1, o2);
+    value exn = caml_do_pending_actions_exn();
+    if (Is_exception_result(exn)) {
+      caml_raise(exn);
+    }
+    CAMLdrop;
+    *v1 = *orig1 = o1;
+    *v2 = *orig2 = o2;
+    *sp = stk->stack = stk->init_stack;
+    return true;
```

Comment:
I do not think there's technically a bug but it is weird to call `CAMLdrop` before using `o1` and `o2`. I suggest replacing it with:
```suggestion
    CAMLreturnT(bool, true);
```